### PR TITLE
Implement related_to:id

### PR DESCRIPTION
**Pull request by jobergum**
Examples

/search/?query=%2Brelated_to:612 %2Bschools

Will do approximate nearest neighboor search using title embedding vector from article 612 and include 'schools's as a matching term. Ranking is done by related-ann-with-matching which is a combination of rawScore(embedding vector) and Vespa's nativeRank. 

Searcher triggers on &id=x, or presence of related_to IntItem in the query (Which is removed before hitting the backend)

### CODE REVIEW

File: vespa-cloud/cord-19-search/src/main/application/searchdefinitions/doc.sd [hunk 1]
```diff
@@ -293,6 +294,23 @@ search doc {
       }
   }
 
+  rank-profile related-ann-with-matching inherits default {
+        function match_score() {
+          expression: nativeRank(title) + nativeRank(abstract)
```
Comment (bratseth) [inline vespa-cloud/cord-19-search/src/main/application/searchdefinitions/doc.sd:299]: normalize: 0.5*nativeRank(title) + 0.5*nativeRank(abstract)

File: vespa-cloud/cord-19-search/src/main/application/search/query-profiles/types/root.xml [hunk 2]
```diff
@@ -1,4 +1,6 @@
 <!-- Copyright Verizon Media. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root. -->
 <query-profile-type id="root" inherits="native">
-  <field name="ranking.features.query(vector)" type="tensor&lt;float&gt;(x[768])" />
+  field name="ranking.features.query(vector)" type="tensor&lt;float&gt;(x[768])" />
```
Comment (freva) [inline vespa-cloud/cord-19-search/src/main/application/search/query-profiles/types/root.xml:3]:
```suggestion
  <field name="ranking.features.query(vector)" type="tensor&lt;float&gt;(x[768])" />
```

Comment (jobergum) [reply vespa-cloud/cord-19-search/src/main/application/search/query-profiles/types/root.xml]: That is serious odd

File: vespa-cloud/cord-19-search/src/main/java/ai/vespa/example/cord19/searcher/RelatedPaperSearcherANN.java [hunk 3]
```diff
@@ -49,46 +51,73 @@ public RelatedPaperSearcherANN(Linguistics linguistics) {
...
+            if(word.getIndexName().equals(RELATED_TO_FIELD)) {
+                return Integer.parseInt(word.getRawWord());
+            }
+        }
+        else if (item instanceof CompositeItem) {
+            CompositeItem compositeItem =(CompositeItem)item;
+            for(Item subItem: compositeItem.items()) {
+                Integer id = traverseQueryTree(subItem);
+                if(id != null) {
+                    compositeItem.removeItem(subItem);
```
Comment (bratseth) [inline vespa-cloud/cord-19-search/src/main/java/ai/vespa/example/cord19/searcher/RelatedPaperSearcherANN.java:112]: I don't think you are allowed to do this while looping over items? You loop with a listIterator instead and call remove on it.

**Pull request merged**